be/src/storage/lake/rowset.cpp

```diff
@@ -233,14 +235,36 @@ Status Rowset::load_segments(std::vector<SegmentPtr>* segments, bool fill_data_c
     size_t footer_size_hint = 16 * 1024;
     uint32_t seg_id = 0;
     bool ignore_lost_segment = config::experimental_lake_ignore_lost_segment;
+    std::vector<std::future<StatusOr<SegmentPtr>>> segment_futures;
+
+    std::unique_ptr<ThreadPool> load_segment_thread_pool;
+    RETURN_IF_ERROR(ThreadPoolBuilder("load_segment")
+                            .set_min_threads(1)
+                            .set_max_threads(config::load_segment_thread_pool_num_max)
+                            .set_max_queue_size(config::load_segment_thread_pool_queue_size)
+                            .set_idle_timeout(MonoDelta::FromMilliseconds(2000))
+                            .build(&load_segment_thread_pool));
+
     for (const auto& seg_name : metadata().segments()) {
-        auto segment_path = _tablet_mgr->segment_location(tablet_id(), seg_name);
-        auto segment_or = _tablet_mgr->load_segment(segment_path, seg_id++, &footer_size_hint, fill_data_cache,
-                                                    fill_metadata_cache, _tablet_schema);
+        auto task = std::make_shared<std::packaged_task<StatusOr<SegmentPtr>()>>([&]() {
+            auto segment_path = _tablet_mgr->segment_location(tablet_id(), seg_name);
+            return _tablet_mgr->load_segment(segment_path, seg_id++, &footer_size_hint, fill_data_cache,
+                                             fill_metadata_cache, _tablet_schema);
+        });
+
+        auto packaged_func = [task]() { (*task)(); };
+        if (auto st = load_segment_thread_pool->submit_func(std::move(packaged_func)); !st.ok()) {
+            return st;
+        }
+        segment_futures.push_back(task->get_future());
+    }
+
+    for (auto& fut : segment_futures) {
+        auto segment_or = fut.get();
         if (segment_or.ok()) {
             segments->emplace_back(std::move(segment_or.value()));
         } else if (segment_or.status().is_not_found() && ignore_lost_segment) {
-            LOG(WARNING) << "Ignored lost segment " << seg_name;
+            LOG(WARNING) << "Ignored lost segment";
```

Comment:
should log file name here, otherwise the log doesn't help anything

Reply:
done